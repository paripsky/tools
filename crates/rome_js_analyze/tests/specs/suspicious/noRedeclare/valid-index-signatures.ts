type ValidIndexSignatures = {
	a: {
		[index: string]: string;
	};
	b: {
		[index: string]: string;
	};
};

function func1(name: string) {
  return {
    name,
  } as { [name: string]: string };
}

function func2(name: string) {
  return {
    name,
  } as { [notName: string]: string };
}
