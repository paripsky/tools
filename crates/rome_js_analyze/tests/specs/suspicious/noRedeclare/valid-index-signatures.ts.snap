---
source: crates/rome_js_analyze/tests/spec_tests.rs
expression: valid-index-signatures.ts
---
# Input
```js
type ValidIndexSignatures = {
	a: {
		[index: string]: string;
	};
	b: {
		[index: string]: string;
	};
};

function func1(name: string) {
  return {
    name,
  } as { [name: string]: string };
}

function func2(name: string) {
  return {
    name,
  } as { [notName: string]: string };
}

```


